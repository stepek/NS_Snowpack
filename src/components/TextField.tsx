import React, {useCallback} from "react"

interface TextFieldProps {
  placeholder?: string
  onChange?: (value: string) => void
  value?: string
}
export function TextField({placeholder, value, onChange}: TextFieldProps) {
  const handleChange = useCallback(
    event => {
      if (onChange) {
        onChange(event.target.value)
      }
    },
    [onChange],
  )

  return (
    <input placeholder={placeholder} onChange={handleChange} value={value} />
  )
}
export default TextField
