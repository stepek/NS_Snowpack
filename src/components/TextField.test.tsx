import React from "react"
import renderer from "react-test-renderer"

import TextField from "./TextField"

test("<TextField> render correctly", () => {
  expect(renderer.create(<TextField />).toJSON()).toMatchSnapshot()
  expect(
    renderer.create(<TextField placeholder={"FooBarBaz"} />).toJSON(),
  ).toMatchSnapshot()
  expect(
    renderer.create(<TextField value={"Foo"} />).toJSON(),
  ).toMatchSnapshot()
})

test("<TextField> call onChange callback", () => {
  const handleChange = jest.fn()
  const component = renderer.create(<TextField onChange={handleChange} />)

  const event = {
    target: {value: "foo"},
  } as React.ChangeEvent<HTMLInputElement>

  component.root.findByType("input").props.onChange(event)

  expect(handleChange).toBeCalledTimes(1)
  expect(handleChange).toBeCalledWith("foo")
})
