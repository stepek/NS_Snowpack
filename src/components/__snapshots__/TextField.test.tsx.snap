// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField> render correctly 1`] = `
<input
  onChange={[Function]}
/>
`;

exports[`<TextField> render correctly 2`] = `
<input
  onChange={[Function]}
  placeholder="FooBarBaz"
/>
`;

exports[`<TextField> render correctly 3`] = `
<input
  onChange={[Function]}
  value="Foo"
/>
`;
